Rover/sailboat.cpp

```diff
@@ -552,3 +575,113 @@ bool Sailboat::motor_assist_low_wind() const
             rover.g2.windvane.wind_speed_enabled() &&
             (rover.g2.windvane.get_true_wind_speed() < sail_windspeed_min));
 }
+
+void Sailboat::update_in_irons()
+{
+    // only enabled for separate mainsail and options bit set
+    {
+        const bool separate_mainsail_channel =
+            (channel_mainsail !=nullptr) &&
+            (channel_mainsail != rover.channel_throttle);
+
+        if (!(separate_mainsail_channel &&
+              option_is_set(OPTION::GET_OUT_OF_IRONS_IN_AUTO))) {
+            clear_in_irons();
+            return ;
+        }
+    }
+
+    const float wind_dir_apparent =
+        degrees(rover.g2.windvane.get_apparent_wind_direction_rad());
+
+    // Already "in irons" or not?
+    if (!_in_irons.state) {
+
+        // check conditions to enter "in irons limbo"
+
+        // being blown backwards by the wind
+        bool  going_backwards = false;
+        {
+            float speed;
+            // only useful if we have a valid speed (e.g fails if no gps fix)
+            if (rover.g2.attitude_control.get_forward_speed(speed)) {
+                going_backwards = is_negative(speed);
+            }
+        }
+
+        // only pure sailboats get into irons
+        const bool cannot_motorsail = this->motor_state == UseMotor::USE_MOTOR_NEVER;
```

Comment:
```suggestion
        const bool cannot_motorsail = motor_state == UseMotor::USE_MOTOR_NEVER;
```

Reply:
Then would suggest to change motor_state to _motor_state to make it distinct from a local variable, or just be explicit, as I have done.